src/ripple/app/misc/AmendmentTableImpl.cpp

```diff
@@ -151,28 +236,27 @@ AmendmentTableImpl::get (std::string const& name)
     return uint256 ();
 }
 
-AmendmentState*
-AmendmentTableImpl::addKnown (const char* amendmentID, const char* friendlyName,
-    bool veto)
+bool
+AmendmentTableImpl::addKnown (uint256 const& hash,
+                              std::string const& friendlyName,
+                              bool veto)
 {
-    uint256 hash;
-    hash.SetHex (amendmentID);
-
     if (hash.isZero ())
     {
         assert (false);
-        return nullptr;
+        return false;
     }
 
-    AmendmentState* f = getCreate (hash, true);
+    ScopedLockType sl (mLock);
+    AmendmentState* s = getCreate (hash, true);
```

Comment:
What?!  A naked pointer?  In C++11 and boost, there is literally never a reason to have a naked pointer!

You want nullability?  Then use `boost::optional` on a reference.  It works exactly the same as a naked pointer _except_ that the naked pointer allows of undefined behavior (since a naked pointer's default initialization gives it an undefined value) and `boost::optional` doesn't.